### PR TITLE
Cleanup the LevelDB residue

### PR DESCRIPTION
**Pull request by rosinL**
As LevelDB support have been remove in Quincy, cleanup the LevelDB residue in this pr.

Signed-off-by: luo rixin <luorixin@huawei.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/man/8/ceph-kvstore-tool.rst
Comment (anthonyeleven) [inline doc/man/8/ceph-kvstore-tool.rst:19]: Please change s/rocksdb/RocksDB

Comment (rosinL) [reply doc/man/8/ceph-kvstore-tool.rst]: I have change all the rocksdb to RocksDB in the comment and docs. Please take a review.

File: doc/radosgw/layout.rst
Comment (cbodley) [inline doc/radosgw/layout.rst:137]: since this documentation is going into main, i don't think it's useful to talk about hammer or quincy at all. can we just remove this text about leveldb/rocksdb? the sentence above says that this distinction isn't important anyway:
> An object's omap is not physically located in the object's storage, but its precise implementation is invisible and immaterial to RADOS Gateway.

Comment (anthonyeleven) [reply doc/radosgw/layout.rst:137]: If it's transparent in the context of RGW, that makes sense.

Comment (rosinL) [reply doc/radosgw/layout.rst]: remove the text has done.

**Comment by rosinL** [comment]
jenkins test make check

**Comment by rosinL** [comment]
jenkins retest this please

**Comment by rosinL** [comment]
jenkins retest this please

**Comment by rosinL** [comment]
jenkins retest this please

**Comment by rosinL** [comment]
@cbodley  @anthonyeleven please take a review.

**Comment by cbodley** [comment]
thanks @rosinL, the rgw changes look great. i'll defer to the core team for approval

**Review by anthonyeleven — COMMENTED** [review]
Left a rewording suggestion

File: qa/tasks/rebuild_mondb.py
Comment (rosinL) [reply qa/tasks/rebuild_mondb.py]: Done

**Comment by rosinL** [comment]
ping @rzarzynski @neha-ojha

File: src/tools/ceph_kvstore_tool.cc
Comment (rzarzynski) [inline src/tools/ceph_kvstore_tool.cc:93]: For a second I was thinking about a check for the `type == "leveldb"` to say it's out but after another one I doubted it makes sense. Thoughts?

File: qa/suites/perf-basic/settings/optimized.yaml
Comment (rzarzynski) [inline qa/suites/perf-basic/settings/optimized.yaml:72]: One less! :-))

File: qa/valgrind.supp
Comment (rzarzynski) [inline qa/valgrind.supp:214]: ACK. Adding this comment to grab attention if something somehow pops up in testing.

File: src/ceph_mon.cc
Comment (rzarzynski) [inline src/ceph_mon.cc:275]: OK, so the least thing the comment is about is `keyring`, right?

File: src/ceph_osd.cc
Comment (rzarzynski) [inline src/ceph_osd.cc:133]: nit: `nullptr`.

File: src/ceph_osd.cc
Comment (rzarzynski) [inline src/ceph_osd.cc:135]: Does RocksDB have any compatibility with these old options? I assume it has not but prefer to be paranoid here (this applies to the other place we're dealing with opts, BTW).

**Review by rzarzynski — COMMENTED** [review]
Overall looks good! Just some nits / questions.

**Comment by rzarzynski** [comment]
Let's link this follow-up PR with the initial one: https://github.com/ceph/ceph/pull/43612.

**Comment by anthonyeleven** [comment]
> @neha-ojha: I wonder about a chance to have this (pretty large...) clean-up in reef. Thoughts?

Aren't we feature-frozen for Reef, and already a month late releasing? Maybe for a dot release when there's more time for validation?

**Comment by rzarzynski** [comment]
Mentioning the LevelDB support in the doc sounds like a doc bug. The PR has some changes I would prefer to bake in `main` but the doc looks backportable. The question is whether they're dissecting. Thinking loudly.

File: src/test/ObjectMap/test_keyvaluedb_atomicity.cc
Comment (rosinL) [reply src/test/ObjectMap/test_keyvaluedb_atomicity.cc:95]: done.

File: src/test/daemon_config.cc
Comment (rosinL) [reply src/test/daemon_config.cc:155]: I hava seach all leveldb-releated options, no option used.

File: src/tools/ceph_kvstore_tool.cc
Comment (rosinL) [reply src/tools/ceph_kvstore_tool.cc:93]: In the help message, there is no leveldb, leveldb is not supported any more, i think is no need to check `type == "leveldb"`

File: src/ceph_mon.cc
Comment (rosinL) [reply src/ceph_mon.cc:275]: sorry, maybe i do not get the point, Do you mean we should make more clear on the comment?

File: src/ceph_osd.cc
Comment (rosinL) [reply src/ceph_osd.cc]: done

File: src/ceph_osd.cc
Comment (rosinL) [reply src/ceph_osd.cc:135]: no, and there do not need the old options.

**Comment by rosinL** [comment]
@rzarzynski Is there anything need to amend, pls take a review.

File: src/test/librados/librados_config.cc
Comment (rzarzynski) [inline src/test/librados/librados_config.cc:35]: ACK. The same set & get pattern as above.

File: src/tools/ceph_kvstore_tool.cc
Comment (rzarzynski) [reply src/tools/ceph_kvstore_tool.cc:93]: I see your point. I was thinking about differentiating the error message: "Unrecognized type" vs "Removed type" but I'm fine with leaving as is.

File: src/ceph_osd.cc
Comment (rzarzynski) [reply src/ceph_osd.cc:135]: ACK, THX.

**Review by rzarzynski — APPROVED** [review]
LGTM!

**Comment by ljflores** [comment]
Rados suite review: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomclyHYQLgL1771-wip-yuri11-testing-2023-05-30-1325

**Pull request merged**
